datafusion/common/src/config.rs

```diff
@@ -1773,7 +1773,7 @@ config_namespace! {
     /// Options controlling JSON format
     pub struct JsonOptions {
         pub compression: CompressionTypeVariant, default = CompressionTypeVariant::UNCOMPRESSED
-        pub schema_infer_max_rec: usize, default = 100
+        pub schema_infer_max_rec: Option<usize>, default = None
```

Comment:
If we make this change to be `Option` then it is inconsistent with the JSON options: 


https://github.com/apache/datafusion/blob/2d7892b6060793e721ce1d66242f485656e0430a/datafusion/common/src/config.rs#L1775-L1776

I think it would be best to keep the two options consistent

Reply:
I did change both `CsvOptions::schema_infer_max_rec` and `JsonOptions::schema_infer_max_rec` so they are consistent. I think the PR body was confusing because of the 2nd change (lifetimes) mentioning CSV. I'll rephrase the PR

Reply:
ah I see -- I was clearly confused